projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/authentication.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from vdk.plugin.control_api_auth.auth_config import InMemAuthConfiguration
+from vdk.plugin.control_api_auth.auth_exception import VDKAuthException
+from vdk.plugin.control_api_auth.base_auth import BaseAuth
+from vdk.plugin.control_api_auth.login_types import LoginTypes
+
+
+class Authentication:
+    """Main class used for authentication."""
+
+    def __init__(
+        self,
+        username: str = None,
+        password: str = None,
+        client_id: str = None,
+        client_secret: str = None,
+        token: str = None,
+        authorization_url: str = None,
+        auth_type: str = None,
+        cache_locally: bool = False,
+    ):
+        """
+        :param username: A user's username in case basic authentication is used.
+        :param password: A user's password in case basic authentication is used.
+        :param client_id:
+            The client identifier;
+            See https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1
+        :param client_secret:
+            The client identifier;
+            See https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1
+        :param token:
+            OAuth api token or a refresh token as specified in
+            https://datatracker.ietf.org/doc/html/rfc6749#section-1.5 and used
+            to obtain access token from the authorization server.
+        :param authorization_url:
+            The URL which exchanges token for access token.
+        :param auth_type:
+            What type of authentication should be used (e.g., refresh_token,
+            basic_auth, etc.).
+        :param cache_locally:
+            A flag, indicating if credentials should be cached locally (in a
+            file).
+        """
+        self._username = username
+        self._password = password
```

Comment:
Are we missing an authentication method? These fields aren't used anywhere; maybe add an empty method to signify that there's code to be added.

Reply:
There will be additional PR-s with the other authentication workflows. I've added a placeholder method.